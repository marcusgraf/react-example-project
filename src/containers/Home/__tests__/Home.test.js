import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { StaticRouter } from 'react-router-dom';

import { Home } from '../index';

describe('<Home />', () => {
  const tree = (props, actions) => renderer.create(
    <StaticRouter context={{}}>
      <Home {...props} {...actions} />
    </StaticRouter>,
  ).toJSON();

  test('should call fetchMoviesIfNeeded when componentDidMount', () => {
    const mockAction = jest.fn();
    const props = {
      home: {},
    };
    const actions = {
      fetchMoviesIfNeeded: mockAction,
    };

    mount(
      <StaticRouter context={{}}>
        <Home {...props} {...actions} />
      </StaticRouter>,
    );

    expect(mockAction).toHaveBeenCalled();
  });

  test('renders the loading status if data invalid', () => {
    const props = {
      home: { readyStatus: 'LOADER_INVALID' },
    };
    const actions = { fetchMoviesIfNeeded: () => {} };

    expect(tree(props, actions)).toMatchSnapshot();
  });

  test('renders the loading status if requesting data', () => {
    const props = {
      home: { readyStatus: 'LOADER_REQUESTING' },
    };
    const actions = { fetchMoviesIfNeeded: () => {} };

    expect(tree(props, actions)).toMatchSnapshot();
  });

  test('renders an error if loading failed', () => {
    const props = {
      home: { readyStatus: 'LOADER_FAILURE' },
    };
    const actions = { fetchMoviesIfNeeded: () => {} };

    expect(tree(props, actions)).toMatchSnapshot();
  });

  test('renders the <MovieCard /> if loading was successful', () => {
    const props = {
      home: {
        readyStatus: 'LOADER_SUCCESS',
        movie: [{ id: '1', name: 'Welly' }],
      },
    };
    const actions = { fetchMoviesIfNeeded: () => {} };

    expect(tree(props, actions)).toMatchSnapshot();
  });
});
