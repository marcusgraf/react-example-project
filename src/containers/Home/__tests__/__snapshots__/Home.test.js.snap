// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders an error if loading failed 1`] = `
<div
  className="Home"
>
  <div
    className="SearchForm"
  >
    <input
      onChange={[Function]}
      placeholder="Search for a movie"
      type="text"
    />
  </div>
  <p>
    Oops, Failed to load list!
  </p>
</div>
`;

exports[`<Home /> renders the <MovieCard /> if loading was successful 1`] = `
<div
  className="Home"
>
  <div
    className="SearchForm"
  >
    <input
      onChange={[Function]}
      placeholder="Search for a movie"
      type="text"
    />
  </div>
  <div
    className="MovieCard"
    style={
      Object {
        "backgroundImage": "url(\\"undefined\\")",
      }
    }
  >
    <div
      className="banner_content_wrapper"
    >
      <div
        className="poster"
      >
        <img
          alt={undefined}
          className="featured_image"
          src={undefined}
        />
        <img
          alt="Play Trailer"
          className="play_button"
          src="https://res.cloudinary.com/dw369yzsh/image/upload/v1470916845/play_button_ngnw1z.png"
        />
      </div>
      <div
        className="content"
      >
        <h2
          className="title"
        />
        <p
          className="description"
        />
        <p
          className="info"
        >
           Views
          <span>
            |
          </span>
          <span>
            |
          </span>
           / 10 Points
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Home /> renders the loading status if data invalid 1`] = `
<div
  className="Home"
>
  <div
    className="SearchForm"
  >
    <input
      onChange={[Function]}
      placeholder="Search for a movie"
      type="text"
    />
  </div>
  <p />
</div>
`;

exports[`<Home /> renders the loading status if requesting data 1`] = `
<div
  className="Home"
>
  <div
    className="SearchForm"
  >
    <input
      onChange={[Function]}
      placeholder="Search for a movie"
      type="text"
    />
  </div>
  <p>
    Loading...
  </p>
</div>
`;
